packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -30,6 +33,55 @@ exports.sourceNodes = async (
     disallowedLinkTypes,
   } = pluginOptions
   const { createNode } = actions
+
+  if (webhookBody && Object.keys(webhookBody).length) {
+    const changesActivity = reporter.activityTimer(
+      `loading Drupal content changes`,
+      {
+        parentSpan,
+      }
+    )
+    changesActivity.start()
+
+    const { secret, action, id, data } = webhookBody
+    if (pluginOptions.secret && pluginOptions.secret !== secret) {
+      reporter.warn(
+        `The secret in this request did not match your plugin options secret.`
+      )
+      return
+    }
+    if (action === `delete`) {
+      actions.deleteNode({ node: getNode(createNodeId(id)) })
+      reporter.log(`Deleted node: ${id}`)
+      return
+    }
+
+    let nodesToUpdate = data
+    if (!(data instanceof Array)) {
```

Comment:
nit: this is more common in javascript than instanceof
```suggestion
    if (!Array.isArray(data)
```